crates/sui/tests/protocol_version_tests.rs

```diff
@@ -593,6 +595,34 @@ mod sim_only_tests {
         assert!(matches!(system_state, SuiSystemState::SimTestV2(_)));
     }
 
+    #[sim_test]
+    async fn check_framework_snapshot_compatibility() {
+        let snapshot_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("../sui-framework-snapshot/bytecode_snapshot");
+        let snapshot_objects: Vec<_> = std::fs::read_dir(&snapshot_path)
+            .unwrap()
+            .flatten()
+            .map(|entry| {
+                let file_name = entry.file_name().to_str().unwrap().to_string();
+                let mut file = std::fs::File::open(snapshot_path.clone().join(file_name)).unwrap();
+                let mut buffer = Vec::new();
+                file.read_to_end(&mut buffer).expect("Unable to read file");
+                let package: MovePackage = bcs::from_bytes(&buffer).unwrap();
+                Object::new_package_from_data(Data::Package(package), TransactionDigest::genesis())
+            })
+            .collect();
+        let test_cluster = TestClusterBuilder::new()
+            .with_epoch_duration_ms(20000)
+            .with_supported_protocol_versions(SupportedProtocolVersions::new_for_testing(
+                START, FINISH,
+            ))
+            .with_objects(&snapshot_objects)
+            .build()
+            .await
+            .unwrap();
+        // TODO: Override using the current framework and let it upgrade.
```

Comment:
I had more of a unit test in mind -- i.e. deserialize the modules into `CompiledModule`s and then run the compatibility check on them.